modules/apps/27-interchain-accounts/host/keeper/msg_server.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/host/types"
+	ibcerrors "github.com/cosmos/ibc-go/v7/modules/core/errors"
+)
+
+var _ types.MsgServer = (*msgServer)(nil)
+
+type msgServer struct {
+	*Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the ICS27 host MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	if keeper.queryRouter == nil {
+		panic("query router must not be nil")
+	}
+	return &msgServer{Keeper: keeper}
+}
+
+// ModuleQuerySafe routes the queries to the keeper's query router if they are module_query_safe.
+// This handler doesn't use the signer.
+func (m msgServer) ModuleQuerySafe(goCtx context.Context, msg *types.MsgModuleQuerySafe) (*types.MsgModuleQuerySafeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	responses := make([][]byte, len(msg.Requests))
+	for i, query := range msg.Requests {
+		var isModuleQuerySafe bool
+		for _, allowedQueryPath := range m.mqsAllowList {
+			if allowedQueryPath == query.Path {
+				isModuleQuerySafe = true
```

Comment:
Yes, but since we are going to bump to go 1.21 and use `slices.Contains` then I will leave it as is. ok?

Reply:
I'd add this anyway in case something goes wrong with 1.21 bump or we decide against the bump. But I'm fine with it either way 🙂

Reply:
I will keep this in mind and if the bump doesn't work out, then I will add it.

Reply:
I think it would make sense for consistency? And go 1.19 is not maintained anymore, so I guess good excuse to bump anyway?

Reply:
I thought that the version in go.mod was only a recommendation (I think I remember that was the conclusion when we had a similar discussion some time ago), but checking again now, it sees like indeed could force users to upgrade. Seems SDK v0.47 is also in 1.19, then we can keep that version. I will make Serdar's suggested change in this PR then.

Reply:
I am wonder though if we should use 1.19 in the workflows as well, just to keep the versions in sync (in case having them out of sync could cause any issues down the road), and fix the broken test in a different way (when I saw it failing I thought of using a different query in the test).